### PR TITLE
make it easier to upload local files

### PR DESCRIPTION
**Pull request by mheilman**
This facilitates uploads from or downloads to local files by allowing users to specify paths rather than file/buffer objects.  This particular type of polymorphism seems pretty well-accepted (cf. [pandas.read_csv](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html)).

### CODE REVIEW

File: civis/io/_files.py [hunk 1]
```diff
@@ -235,6 +237,13 @@ def file_to_civis(buf, name, api_key=None, client=None, **kwargs):
     Small or non-seekable file-like objects will be uploaded with a
     single post.
     """
+    if isinstance(buf, str):
+        with open(buf, 'rb') as f:
+            return _file_to_civis(
+                f, name, api_key=api_key, client=client, **kwargs)
```
Comment (keithing) [inline civis/io/_files.py]: What happens when `buf` is not a `str` here?

**Comment by keithing** [comment]
I like this.  I think polymorphism is okay here.

**Comment by stephen-hoover** [comment]
Our current read/write functions resist polymorphism, which is like the standard library functions (e.g. `pickle.load`). On the other hand, reading from files on disk is incredibly common, and third-party libraries I've seen (e.g. `joblib` and `pandas`) are polymorphic in exactly the way this PR proposes. I think this PR is a good idea. I know I've written this wrapper myself more than once.

**Comment by mheilman** [comment]
Haha oops.  I'll fix that!

File: civis/io/_files.py (hunk 1 above)
Comment (mheilman) [reply civis/io/_files.py]: This should be fixed now.

File: civis/io/_files.py [hunk 2]
```diff
@@ -235,6 +238,16 @@ def file_to_civis(buf, name, api_key=None, client=None, **kwargs):
     Small or non-seekable file-like objects will be uploaded with a
     single post.
     """
+    if isinstance(buf, str):
```
Comment (stephen-hoover) [inline civis/io/_files.py]: You should check against `six.string_types` instead of `str`, for Python 2 compatibility.

File: civis/io/_files.py [hunk 3]
```diff
@@ -290,9 +303,26 @@ def civis_to_file(file_id, buf, api_key=None, client=None):
...
     >>> with open("my_file.txt", "wb") as f:
     ...    civis_to_file(file_id, f)
+    >>> # Download a file as a bytes object.
+    >>> import io
+    >>> buf = io.BytesIO()
+    >>> civis_to_file(file_id, buf)
+    >>> # Note that s could be converted to a string with s.decode('utf-8').
+    >>> s = buf.read()
     """
+    if isinstance(buf, str):
```
Comment (stephen-hoover) [inline civis/io/_files.py]: The check should be on `six.string_types`.

**Comment by stephen-hoover** [comment]
LGTM!

**Pull request merged**